single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/SingleLineDiagram.java

```diff
@@ -174,138 +128,71 @@ public static void draw(Graph graph, Path svgFile, LayoutParameters layoutParame
-        drawSubstation(network, id, writerForSvg, metadataWriter, layoutParameters, componentLibrary,
-                new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
-                createDefaultStyleProvider(network),
-                "");
+        voltageLevelLayoutFactory.create(voltageLevelGraph).run(layoutParameters);
+        DefaultSVGWriter svgWriter = new DefaultSVGWriter(param.getComponentLibrary(), param.getLayoutParameters(), param.getSvgParameters());
+        draw(voltageLevelGraph, writerForSvg, metadataWriter, svgWriter, param.createLabelProvider(network), param.createStyleProvider(network), param.getSvgParameters().getPrefixId());
     }
 
-    public static void drawSubstation(Network network, String id, Writer writerForSvg, Writer metadataWriter, LayoutParameters layoutParameters, ComponentLibrary componentLibrary,
-                                      DiagramLabelProvider initProvider, StyleProvider styleProvider, String prefixId) {
-        drawSubstation(network, id, writerForSvg, metadataWriter, layoutParameters, componentLibrary,
-                new HorizontalSubstationLayoutFactory(), new SmartVoltageLevelLayoutFactory(network),
-                initProvider, styleProvider, prefixId);
-    }
+    public static void drawSubstation(Network network, String substationId, Writer writerForSvg, Writer metadataWriter, Param param) {
+        LayoutParameters layoutParameters = param.getLayoutParameters();
+        VoltageLevelLayoutFactory voltageLevelLayoutFactory = param.getVoltageLevelLayoutFactory();
+        SubstationLayoutFactory substationLayoutFactory = param.getSubstationLayoutFactory();
 
-    private static void drawSubstation(Network network, String substationId, Writer writerForSvg, Writer metadataWriter, LayoutParameters layoutParameters, ComponentLibrary componentLibrary,
-                                       SubstationLayoutFactory sLayoutFactory, VoltageLevelLayoutFactory vLayoutFactory,
-                                       DiagramLabelProvider initProvider, StyleProvider styleProvider, String prefixId) {
-        Objects.requireNonNull(sLayoutFactory);
-        Objects.requireNonNull(vLayoutFactory);
+        Objects.requireNonNull(substationLayoutFactory);
+        Objects.requireNonNull(voltageLevelLayoutFactory);
 
         SubstationGraph substationGraph = new NetworkGraphBuilder(network).buildSubstationGraph(substationId);
-        sLayoutFactory.create(substationGraph, vLayoutFactory).run(layoutParameters);
-        draw(substationGraph, writerForSvg, metadataWriter, layoutParameters, componentLibrary, initProvider, styleProvider, prefixId);
-    }
-
-    public static void draw(Graph graph, Writer writerForSvg, Writer metadataWriter, LayoutParameters layoutParameters, ComponentLibrary componentLibrary,
-                            DiagramLabelProvider initProvider, StyleProvider styleProvider, String prefixId) {
```

Comment:
You removed this method, but it was used in unit tests. Removing it forced you to instantiate a DefaultSvgWriter in all the unit tests. We don't want this, as this leads to always the same code in all the tests.

With the new design the equivalent would be a `draw(graph, writerForSvg, metadataWriter, param)` method.